### ISSUE TITLE
new-parens gives false warning when passing function param to constructor

### ISSUE THREAD

**Issue by mamacdon** [eslint/eslint#82]
Run ESLint on this code:

``` javascript
var foo = new Thing(function() { });
```

You'll get a warning from the new-parens rule:

```
line 1, col 10, Warning - Missing '()' invoking a constructor
```

I expected no warning, as the constructor call is properly parenthesized. 

This issue occurs when a function (or an object literal containing a function) is among the params passed to the constructor.
